### DISCUSSION TITLE
childImageSharp is null for filetypes supported by sharp

### DISCUSSION THREAD

**Discussion by olivier-lacroix** [gatsbyjs/gatsby#28719] [Ideas / Feature Requests]
## Description

Hi there,

I am willing to generate thumbnails for PDF and use them with gatsby-image. Sharp can handle and convert PDFs just fine (when libvips is compiled with PDF support). However, even when setting `checkSupportedExtensions: false` in gatsby-transformer-sharp, childImageSharp returns null.

As a workaround, I manually create an intermediary PNG image (using sharp) in a File Node, that can then be fed to gatsby-transformer-sharp. Ideally, this intermediary step wouldn't be required.

### Steps to reproduce

Request childImageSharp on a file Node pointing to a PDF file

### Expected result

gatsby-transformer-sharp would handle gracefully any format supported by sharp

### Actual result

null is returned

### Environment

System:
    OS: Linux 4.19 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (4) x64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 15.4.0 - /usr/bin/node
    npm: 7.0.15 - /usr/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Firefox: 78.5.0esr
  npmPackages:
    gatsby: ^2.24.91 => 2.29.1 
    gatsby-image: ^2.4.21 => 2.8.0 
    gatsby-plugin-manifest: ^2.4.37 => 2.9.0 
    gatsby-plugin-netlify-cms: ^4.6.0 => 4.7.0 
    gatsby-plugin-offline: ^3.2.38 => 3.7.0 
    gatsby-plugin-optimize-svgs: ^1.0.4 => 1.0.4 
    gatsby-plugin-postcss: ^3.0.4 => 3.4.0 
    gatsby-plugin-react-helmet: ^3.3.14 => 3.7.0 
    gatsby-plugin-react-i18next: 0.0.27 => 0.0.27 
    gatsby-plugin-sharp: ^2.6.43 => 2.11.1 
    gatsby-source-filesystem: ^2.3.37 => 2.8.0 
    gatsby-transformer-remark: ^2.11.0 => 2.13.0 
    gatsby-transformer-sharp: ^2.5.20 => 2.9.0 
  npmGlobalPackages:
    gatsby-cli: 2.15.1

**Comment by olivier-lacroix** [comment]
@KyleAMathews , in response to your message on the original bug report, my understanding looking at the code, is that this option is used to bypass an early exit ( see [here](https://github.com/gatsbyjs/gatsby/blob/10dfe011c368e28e8de19f0f5569748ebb9a6bc3/packages/gatsby-transformer-sharp/src/create-resolvers.js#L13) ). However, it is not used [here](https://github.com/gatsbyjs/gatsby/blob/001e045c92dfd37a7d573d75ee9d81ae44c8e7a5/packages/gatsby-transformer-sharp/src/on-node-create.js#L4); maybe it should?

By adding pdf to the supportedExtensions variable, childImageSharp does not return null anymore: it crashes with `"ProbeError: unrecognized file format"`. probe-image-size crashes, which kind of makes sense, PDF being a vector format.

Not sure what would be the preferred way to handle that.

**Reply by KyleAMathews** [reply]
Maybe just an option like `addToSupportedExtensions` (probably a better name out there...) where you can arbitrary additional extensions. It seems like there'd need to be additional changes done first though...

Another option is a transformer that first extracts the images out of PDFs as then transformer-sharp could then handle them normally.
